### ISSUE TITLE
[PERF] Lots of "Not responding" / Spinning wheels with large js file

### ISSUE THREAD

**Issue by jnachtigall** [devtools-html/debugger.html#1377]
I know this isn't the most useful bug report, but I am getting lots of "Not responding" issues on Windows 10. That is, the Firefox window regularly hangs for about 5-10secs. Happens:
- When I simply Ctrl+F in a JS File which is 2MB
- When I have breakpoints in this file and reload

My system isn't "weak" either: 
* Intel(R) Core(TM) i5-3570 CPU @ 3.40GHz, 3401 MHz, 4 Kern(e), 4 logische(r) Prozessor(en)
* 12 GB RAM

Really hard getting any work done with current Firefox Developer Editon I am afraid to say :-/ Let me know if I could / should run any diagnostics/profiling or whatever that might help you...

**Comment by jasonLaster** [comment]
@jnachtigall that is *quite* bad. 

* I'll add a quick 5mb file example so that it is easy for others to test this
* Would it be possible for you to reach out me at jlaster at mozilla <dot> com? It'd be nice to get some more information here

@clarkbw, @Garbee, @viswanathamsantosh mind investigating as well?

**Comment by jnachtigall** [comment]
@jasonLaster I'll look into this tomorrow. And in case I cannot reproduce with your 5mb.js file then I'll try to send you a minimal working example by mail.

**Comment by viswanathamsantosh** [comment]
Strange! I cannot see the 5mb.js file in the source panel. However i was able to find the jquery.js file. Regarding the browser not responding, I always face this issue when i open a tab to debug irrespective of the size of js file.

System configuration is Intel(R) Core i5-4200U CPU @ 1.6Ghz 2.30 GHz with 6:00GB RAM.

**Comment by jnachtigall** [comment]
>  I cannot see the 5mb.js file in the source panel.

@viswanathamsantosh Me neither. I had this reported in https://github.com/devtools-html/debugger.html/pull/1384#issuecomment-265435911 (Sorry, should have shared before) @jasonLaster maybe open a separate bug for the large file not being visible in the source panel?

**Comment by jnachtigall** [comment]
I only have the Performance problems on large files. I cannot reproduce in the exteme form I had when I originally posted this bug ( @viswanathamsantosh maybe can?) , but here's a here's Call Tree screenshot for when having a concatenated 2MB js:

STR:
1. Set breakpoint at end of the js file which make the editor jump to this position when opening
2. Reload
2. Start Performance Recording in Browser Toolbox
3. Browser freezes
4. Stop Recording

![2016-12-09 14_58_31- kartenausschnitt-2 importiert -2 0 rgb-farben 1 ebene 1240x640 gimp](https://cloud.githubusercontent.com/assets/3307404/21051379/4277a4b2-be20-11e6-8389-18ad8904cee2.png)

TBH, Firefox 50 with the old debugger also hangs. Chrome debugger is also does not feel fast, but at least does not go into Windows' "Not Responding" mode. 

Actually, just scrolling or opening at the middle/end (if you have a previously remembered pos) in a large js file makes Firefox hang. 

And here's a screenshot for stepping over a breakpoint 5 times (which also feels slow) in a large file:

![2016-12-09 15_18_49- kartenausschnitt-2 importiert -2 0 rgb-farben 1 ebene 1240x640 gimp](https://cloud.githubusercontent.com/assets/3307404/21051949/d504323a-be22-11e6-8d7d-44fce4d480df.png)
(I don't know what Startup::XRE_Main is so not sure how useful this is)

**Comment by yurydelendik** [comment]
Looks like https://devtools-html.github.io/debugger-examples/examples/large-files.html is broken new. Can somebody provide a test case that reproduces the problem with STR?

**Comment by jasonLaster** [comment]
Working on the debugger and sensor web is fast enough after some speed ups that i think we can close this and open a new issue with a better STR

https://devtools-html.github.io/debugger-examples/

**Comment by jnachtigall** [comment]
I'm still having this with today's Nightly:
![hanging](https://cloud.githubusercontent.com/assets/3307404/23456281/19deeed2-fe73-11e6-8f86-f4af4a105938.png)
TBH, I think this is more of a CodeMirror bug than something that is actionable for Debugger.html.

Anyway, I'll email a locally saved file to @jasonLaster

**Comment by jasonLaster** [comment]
@jnachtigall sorry, unfortunately nightly isn't up to date yet. 
waiting on this merge: https://bugzilla.mozilla.org/show_bug.cgi?id=1338567

**Comment by jnachtigall** [comment]
@jasonLaster Yeah, I know, but I am pretty confident that the "Unresponsive script" will be also present in the current debugger version.  Anyway, I've sent you a mail. As said I think it's more likely a CodeMirror issue, but maybe Debugger could catch this more gracefully...

I agree this should not be blocker and is hard to take action upon (except maybe a report to CodeMirror)

BTW, even when https://bugzilla.mozilla.org/show_bug.cgi?id=1338567 is merged, then Nightly will have - talking of now - the version as of 10th Feb and again already 3 weeks out of sync :-/

**Comment by jasonLaster** [comment]
^ nope this version was as of tuesday

**Comment by jnachtigall** [comment]
@jasonLaster ok, sorry, then I misread. Thanks for clarifying. Looking forward 👍

**Comment by jasonLaster** [comment]
@tromey @gregtatum could you share a profile you created with some large files?

**Comment by brabadu** [comment]
Hey!
I'm having similar issues. I have huge js-file (100k loc), and it's really painful to debug in new debugger. While Chrome devtools works absolutely normal with this file.

I've made performance profile of debugger.html, captured in chrome for just one step of debugger. 
[chrome-debugger.html-profile.json.zip](https://github.com/devtools-html/debugger.html/files/1232542/chrome-debugger.html-profile.json.zip)

There's a lot of calls of `utf8decode`, and later `removeLineClass` for CodeMirror.

**Comment by jasonLaster** [comment]
Thanks! I appreciate it

**Comment by codehag** [comment]
I am closing this in favor of #3460 (duplicate issue) Will refer back to this when working on it
